options.go

```diff
@@ -0,0 +1,453 @@
+//
+// Copyright (c) 2025 The go-yaml Project Contributors
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package yaml
+
+import (
+	"errors"
+
+	"go.yaml.in/yaml/v4/internal/libyaml"
+)
+
+// LineBreak represents the line ending style for YAML output.
+type LineBreak = libyaml.LineBreak
+
+// Line break constants for different platforms.
+const (
+	LNBreak   = libyaml.LN_BREAK   // Unix-style \n (default)
+	CRBreak   = libyaml.CR_BREAK   // Old Mac-style \r
+	CRLNBreak = libyaml.CRLN_BREAK // Windows-style \r\n
+)
+
+// options holds configuration for both loading and dumping YAML.
+type options struct {
+	// Loader options
+	singleDocument bool
+	knownFields    bool
+	uniqueKeys     bool
+
+	// Dumper options
+	indent                int
+	compactSeqIndent      bool
+	lineWidth             int
+	unicode               bool
+	canonical             bool
+	lineBreak             LineBreak
+	explicitStart         bool
+	explicitEnd           bool
+	flowSimpleCollections bool
+
+	// Plugin options
+	plugins []Plugin
+}
+
+// Option allows configuring YAML loading and dumping operations.
+type Option func(*options) error
+
+// WithIndent sets the number of spaces to use for indentation when
+// dumping YAML content.
+//
+// Valid values are 2-9. Common choices: 2 (compact), 4 (readable).
+func WithIndent(indent int) Option {
+	return func(o *options) error {
+		if indent < 2 || indent > 9 {
+			return errors.New("yaml: indent must be between 2 and 9 spaces")
+		}
+		o.indent = indent
+		return nil
+	}
+}
+
+// WithCompactSeqIndent configures whether the sequence indicator '- ' is
+// considered part of the indentation when dumping YAML content.
+//
+// If compact is true, '- ' is treated as part of the indentation.
+// If compact is false, '- ' is not treated as part of the indentation.
+// When called without arguments, defaults to true.
+func WithCompactSeqIndent(compact ...bool) Option {
```

Comment:
This is use of variadic for one arg only, sounds strange to me in Go.

@alexandear @brackendawson @chavacava

We expect variadic to be n-sized not fixed one.